scripts/reg.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# create a directory to store user passwords
+mkdir -p ./scripts/registry_auth
+
+# use htpasswd to create an encrypted file
+htpasswd -Bbn test 1234 > ./scripts/registry_auth/htpasswd
+
+# check if there is a container named registry
+if [ "$(docker ps -aq -f name=registry)" ]; then
+    # stop and remove the container named registry
+    docker stop registry
+    docker rm registry
+fi
+
+# start the Docker Registry with authentication
+docker run -p 5001:5000 \
+--restart=always \
+--name registry \
+-v /var/lib/registry:/var/lib/registry \
+-v $PWD/scripts/registry_auth/:/auth/ \
+-e "REGISTRY_AUTH=htpasswd" \
+-e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" \
+-e "REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd" \
+-d registry
```

Comment:
Actually, I recommend using zot instead of registry, [https://github.com/project-zot/zot] is a CNCF project for OCI-native container image registry and it has no historical baggage. on the other hand, oras also running the ci with zot.
But kpm is using the registry, so i just copy it to here.

Reply:
BTW, i'm also using zot to running with CI for KCL package, https://github.com/liangyuanpeng/kind-kcl/pull/1/files#diff-13b39b8636488e063a5bb3d00621a45fc69a1ddbbefd4a3d3a078644824d4213R6